server/db/db.js

```
@@ -4,123 +4,190 @@
 
 'use strict';
 
-var createHash = require('crypto').createHash;
-var log = require('../logger')('server.db.db');
-var utils = require('./utils');
+var crypto = require('crypto');
 
-var pool = utils.createPool();
+var pg = require('pg');
+var uuid = require('uuid');
 
-function _verbose(funcName, funcArgs) {
-  var args = funcArgs && [].join.call(funcArgs, ', ');
-  log.verbose(funcName + ' called. arguments are: ' + args);
-}
+var config = require('../config');
+var log = require('../logger')('server.db.db');
+var dbParams = {
+  user: config.get('db.postgres.user'),
+  password: config.get('db.postgres.password'),
+  host: config.get('db.postgres.host'),
+  port: config.get('db.postgres.port'),
+  database: config.get('db.postgres.database'),
+  ssl: config.get('db.postgres.ssl')
+};
+var _verbose = function() {
+  var logline = [].join.call(arguments, ', ');
+  log.verbose(logline);
+};
 
 // DB API uses callbacks for the moment; TODO return promises?
 // TODO validate DB inputs from the API at some point
-module.exports = {
-  createUser: function(fxaId, email, oauthToken, cb) {
-    _verbose('db.createUser', arguments);
-    var query = 'INSERT INTO users (fxaId, email, oauthToken) ' +
-                'VALUES (?, ?, ?)';
-    pool.query(query, [fxaId, email, oauthToken], function(err) {
-      if (err) {
-        log.warn('error saving user: ' + err);
-        // TODO send the item to a retry queue?
-        return cb(err);
-      }
-      _verbose('db.createUser: created user');
-      cb(err);
+// TODO if we get an unrecoverable error from a connection, we want to kill it,
+// so that it is not returned to the pool (hapi-node-postgres suggestion).
+var user = {
+  create: function(fxaId, email, oauthToken, cb) {
+    _verbose('db.user.create', fxaId, email, oauthToken);
+    var query = 'INSERT INTO users (fxaId, email, oauthToken, createdAt) ' +
+                'VALUES ($1, $2, $3, $4)';
+    pg.connect(dbParams, function(err, client, done) {
```

We aren't handing `err` here. We may want to fail fast-n-hard.